### ISSUE TITLE
How to define and handle webhooks

### ISSUE THREAD

**Issue by giorbismiguel** [brezzhnev/atlassian-connect-core#1]
Hello,

I need to listen when the user creates and modify a page in Confluence. Too I
would like to know if with your package a can do request to the API of
Confluence.

In your page of the package only I can see one example but I don't understand.

Can you help me send me some examples about there?

Thanks

**Comment by brezzhnev** [comment]
Hi @giorbismiguel,

Got it. You want to implement webhooks, but the package didn't support a simplified way to manage it. It's in my plans yet.

You always can implement it ordinarily:

1. Create controller and action to handle incoming webhooks, define route (POST method)

1. Define the needed webhooks in the descriptor
```PHP
Descriptor::base()
    ...
    ->withModules([
        'webhooks' => [
           [
              'event' => 'page_updated',
              'url' => route('webhook-handler-named-route')
           ],
           ...
        ]
    ]);
```

After add-on reinstalling you can handle incoming requests in your defined action.

**Comment by giorbismiguel** [comment]
Thanks, but when I create the controller and action to handle incoming webhooks:  
file: route/web.php
Route::get('/home', 'HomeController@index')->name('home');

file: app\Providers\AppServiceProvider.php
Descriptor::base()
     ->withModules([
        'webhooks' => [
                    [
                        'event' => 'page_updated',
                        'url' => route('home')
                    ]
        ]
]);

and test with ngrok, the navigator shows me:

![image](https://user-images.githubusercontent.com/2695637/30709871-ac1646e8-9ed1-11e7-94ac-bb86776577dc.png)

**Comment by giorbismiguel** [comment]
I resolved the problem change the position of the providers in the file: conf/app.php

Before
App\Providers\AuthServiceProvider::class,
// App\Providers\BroadcastServiceProvider::class,
App\Providers\EventServiceProvider::class,
App\Providers\AppServiceProvider::class,
App\Providers\RouteServiceProvider::class,

After
App\Providers\RouteServiceProvider::class,
App\Providers\AuthServiceProvider::class,
// App\Providers\BroadcastServiceProvider::class,
App\Providers\EventServiceProvider::class,
App\Providers\AppServiceProvider::class,

**Comment by giorbismiguel** [comment]
How do you manage the request in the action?

**Comment by brezzhnev** [comment]
@giorbismiguel great, I'll add a note to the docs regarding this.

Btw, if you are using Laravel 5.5 you don't need to register provider or alias for this package.

Regarding managing the request, I created a "gateway" class for this, there pass Request instance and event name, then I use mapping of events and methods and for each of my supported event I created appropriate method, eg. for the event `jira:issue_updated` method will be `issueUpdated`.
